pkg/scheduler/util/utils.go

```diff
@@ -109,3 +108,66 @@ func GetPodAntiAffinityTerms(affinity *v1.Affinity) (terms []v1.PodAffinityTerm)
 	}
 	return terms
 }
+
+// GetPodCondition extracts the provided condition from the given status and returns that.
+// Returns nil and -1 if the condition is not present, and the index of the located condition.
+func GetPodCondition(status *v1.PodStatus, conditionType v1.PodConditionType) (int, *v1.PodCondition) {
+	if status == nil {
+		return -1, nil
+	}
+	return GetPodConditionFromList(status.Conditions, conditionType)
+}
+
+// GetPodConditionFromList extracts the provided condition from the given list of condition and
+// returns the index of the condition and the condition. Returns -1 and nil if the condition is not present.
+func GetPodConditionFromList(conditions []v1.PodCondition, conditionType v1.PodConditionType) (int, *v1.PodCondition) {
+	if conditions == nil {
+		return -1, nil
+	}
+	for i := range conditions {
+		if conditions[i].Type == conditionType {
+			return i, &conditions[i]
+		}
+	}
+	return -1, nil
+}
+
+// UpdatePodCondition updates existing pod condition or creates a new one. Sets LastTransitionTime to now if the
+// status has changed.
+// Returns true if pod condition has changed or has been added.
+func UpdatePodCondition(status *v1.PodStatus, condition *v1.PodCondition) bool {
```

Comment:
@damemi I am worried about this function, it is not trivial, and so I am afraid we will diverge from the canonical way of updating pod status. As we discussed before, what we want is to get those functions into staging first.

Reply:
@ahg-g you are right, I thought this function had been *moved* to our utils (along with the others), but I didn't realize it was actually *copied*. Looking at this now I see that several of these functions are duplicated, which I agree is not what we want. Sorry about that.

Where in staging do you think it would be appropriate to move these functions? Maybe apimachinery?

/lgtm cancel